components/post_priority/post_priority_picker.tsx

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, memo} from 'react';
+import {useIntl} from 'react-intl';
+import classNames from 'classnames';
+
+import {AlertOutlineIcon, AlertCircleOutlineIcon, MessageTextOutlineIcon, CheckIcon} from '@mattermost/compass-icons/components';
+
+import {PostPriority} from '@mattermost/types/posts';
+
+import Label, {LabelType} from 'components/label/label';
+import menuItem from 'components/widgets/menu/menu_items/menu_item';
+
+import './post_priority_picker.scss';
+
+type Props = {
+    priority?: PostPriority;
+    onClose: () => void;
+    onApply: (props: {priority: PostPriority|undefined}) => void;
+    placement: string;
+    rightOffset?: number;
+    topOffset?: number;
+    leftOffset?: number;
+    style?: React.CSSProperties;
+}
+
+type ItemProps = {
+    isSelected: boolean;
+    onClick: () => void;
+    ariaLabel: string;
+    text: React.ReactNode;
+    id: string;
+}
+
+function Item({
+    onClick,
+    ariaLabel,
+    text,
+    isSelected,
+    id,
+}: ItemProps) {
+    return (
+        <button
+            id={id}
+            aria-label={ariaLabel}
+            className='style--none PostPriorityPicker__item'
+            onClick={onClick}
+        >
+            {text && <span className='MenuItem__primary-text'>{text}</span>}
+            {isSelected && (
+                <span className='PostPriorityPicker__check'>
+                    <CheckIcon size={18}/>
+                </span>
+            )}
+        </button>
+    );
+}
+
+const MenuItem = menuItem(Item);
+
+function PostPriorityPicker({
+    priority,
+    onClose,
+    onApply,
+    placement,
+    rightOffset = 0,
+    topOffset = 0,
+    leftOffset = 0,
+    style,
+}: Props) {
+    const {formatMessage} = useIntl();
+
+    const handleSelect = useCallback((type?: PostPriority) => () => {
+        onApply({priority: type});
+        onClose();
+    }, []);
```

Comment:
isn't eslint complaining here because of the `exhaustive-deps` rule?

Reply:
nope... but I'll make the changes 😅 thanks